atoma-state/src/state_manager.rs

```diff
@@ -284,6 +284,48 @@ impl AtomaState {
             .collect()
     }
 
+    /// Get task for model with the cheapest price (based on the current node subscription).
+    ///
+    /// This method fetches the task from the database that is associated with
+    /// the given model through the `tasks` table and has the cheapest price per compute unit.
+    /// The price is determined based on the node subscription for the task.
+    ///
+    /// # Arguments
+    ///
+    /// * `model` - The model name for the task.
+    ///
+    /// # Returns
+    ///
+    /// - `Result<Option<CheapestTask>>>`: A result containing either:
+    ///  - `Ok(Some<CheapestTask>)`: A `CheapestTask` object representing the task with the cheapest price.
+    ///  - `Ok(None)`: If no task is found for the given model.
+    ///  - `Err(AtomaStateManagerError)`: An error if the database query fails or if there's an issue parsing the results.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if the database query fails.
+    #[instrument(level = "trace", skip_all, fields(%model))]
+    pub async fn get_cheapest_task_for_model(&self, model: &str) -> Result<Option<CheapestTask>> {
```

Comment:
This naming is erroneous. You should pick the cheapest stack currently available, with a given selected node. A task is not price bound.

Reply:
renamed to `get_cheapest_node_for_model`